core/src/main/scala/cats/Inject.scala

```diff
@@ -0,0 +1,54 @@
+package cats
+
+/**
+ * Inject is a type class to inject a value of type `A` into a
+ * coproduct of types `B` when `A :≺: B`. The coproduct `B` is
+ * modelled with `Either`.
```

Comment:
@edmundnoble I like your explanation. I tweaked it a tad-- how about:
```
/**
 * Inject is a type class providing an injection from type `A` into
 * another type `B`.  An injection is a function `inj` which does not
 * destroy any information: for every `b: B` there is at most one
 * `a: A` such that `inj(a) = b`.
 *
 * Because of this all injections admit partial inverses `prj` which
 * pair a value `b: B` back with a single value `a: A`.
 *
 * @since 1.0
 * @note Prior to cats 1.0, Inject handled injection for type
 * constructors. For injection of type constructors, use [[InjectK]].
 *
 * @see [[InjectK]] for injection for [[cats.data.EitherK]]
 */
abstract class Inject[A, B] // ...
```

I'll give the same treatment to `InjectK` once proper doc is decided.